python/tk_multi_breakdown2/utils.py

```diff
@@ -47,7 +57,42 @@ def get_ui_published_file_fields(app):
     fields += utils.resolve_sg_fields(file_history_config.get("top_left"))
     fields += utils.resolve_sg_fields(file_history_config.get("top_right"))
     fields += utils.resolve_sg_fields(file_history_config.get("body"))
-    if file_history_config["thumbnail"] and "image" not in fields:
-        fields.append("image")
+    if file_history_config["thumbnail"]:
+        if "image" not in fields:
+            fields.append("image")
+        # We need the linked Version's image if setting
+        # use_version_thumbnail_as_fallback is enabled.
+        if "version.Version.image" not in fields:
+            fields.append("version.Version.image")
 
     return list(set(fields))
+
+
+def get_item_image_field(item):
+    """
+    Get the field to use for the thumbnail for the given item.
+
+    It returns "image" if the item has an image field and there's an
+    actual image to use. If there's no image field or no image to use,
+    it tries to use the version.Version.image field in the same way,
+    if setting use_version_thumbnail_as_fallback is enabled.
+
+    If there's no image field or no image to use, it returns None.
+
+    :param item: The QStandardItem to get the thumbnail field for.
+    :returns: A SG field, e.g. "image", or ``None``.
+    """
+    sg_data = item.sg_data
+    app = sgtk.platform.current_bundle()
+    if not sg_data:
+        return None
+    use_version_thumbnail_as_fallback = app.get_setting("use_version_thumbnail_as_fallback")
+    # When it's an empty thumbnail, it's an AWS link with a "no_preview_t.jpg" image.
+    if sg_data.get("image") and "no_preview_t.jpg" not in sg_data.get("image"):
```

Comment:
So if I upload a "no_preview_t.jpg" image, this test will wrongly think there is no thumbnail?

Reply:
yes but I don't know of a better way of dealing with this...

Reply:
Well may be check it is an AWS url, anyway, not a big thing.